simulator/infra/cache/cache_tag_array.cpp

```diff
@@ -1,174 +1,73 @@
+    if ( size == 0)
         serr << "ERROR: Wrong arguments! Cache size should be greater than zero"
              << std::endl << critical;
-
+    
     if ( ways == 0)
         serr << "ERROR: Wrong arguments! Num of ways should be greater than zero"
              << std::endl << critical;
-
-    if ( line_size == 0)
-        serr << "ERROR: Wrong arguments! Line size should be greater than zero"
+    
+    if ( size < ways)
+        serr << "ERROR: Wrong arguments! Cache size should be greater"
+             << "than the number of ways" 
              << std::endl << critical;
 
-    if ( addr_size_in_bits == 0)
-        serr << "ERROR: Wrong arguments! Address size should be greater than zero"
+    if ( !is_power_of_two( size))
+        serr << "ERROR: Wrong argumets! Cache size should be a power of 2"
+             << std::endl << critical;         
+    
+    if ( size % ways != 0)
+        serr << "ERROR: Wrong arguments! Cache size should be multiple of"
+             << "the number of ways"
              << std::endl << critical;
 
-    /*
-     * It also checks "size_in_bytes < line_size" and "size_in_bytes
-     * < ways".
-     */
-    if ( size_in_bytes / ways < line_size)
-        serr << "ERROR: Wrong arguments! Size of each way should be not "
-                  << "less than size of block (size in bytes of cache should "
-                  << "be not less than number of ways and size of block in "
-                  << "bytes)." << critical;
-
-    /*
-     * It also checks "size_in_bytes % line_size != 0" and
-     * "size_in_bytes % ways != 0".
-     */
-    if ( ( size_in_bytes % ( line_size * ways)) != 0)
-        serr << "ERROR: Wrong arguments! Size of cache should be a "
-                  << "multiple of block size in bytes and number of ways."
-                  << critical;
-
-    /* The next two use: "2^a=b"<=>"b=100...000[2]"<=>"(b&(b-1))=0". */
-    if ( !is_power_of_two( size_in_bytes / ( line_size * ways)))
-        serr << "ERROR: Wrong arguments! Number of sets should be a power"
-                  << " of 2." << critical;
-
-    if ( !is_power_of_two( line_size))
-        serr << "ERROR: Wrong arguments! Block size should be a power of "
-                  << "2." << critical;
+    return size / ways;
 }
 
-CacheTagArray::CacheTagArray( uint32 size_in_bytes,
-                              uint32 ways,
-                              uint32 line_size,
-                              uint32 addr_size_in_bits) :
-                              CacheTagArrayCheck( size_in_bytes, ways,
-                                                  line_size, addr_size_in_bits),
-                              num_sets( size_in_bytes / ( line_size * ways)),
-                              array( num_sets, std::vector<CacheTag>( ways)),
-                              lru( ways, num_sets)
-{
 
-}
 
-std::pair<bool, uint32> CacheTagArray::read( Addr addr)
+std::pair<Addr, uint32> CacheTagArray::read( Addr addr)
```

Comment:
What does Addr mean here?  There are only two results possible: cache hit (true) and cache miss (false).

Reply:
Sorry, It's definitely a mistake